algorithms/prims-algorithm/prims-algorithm.tex

```diff
@@ -0,0 +1,63 @@
+\section{Prim's Algorithm}\index{graph!Prim's Algorithm}
+#ifdef hackpackpp
+Prims algorithm is a greedy algorithm that allows for finding a minimum spanning tree/forest in $O(E + \log{V})$ time.
+Big picture, Prims algorithm works by doing the following:
+
+\begin{enumerate}
+	\item Choose an arbitrary unvisited node in the graph.
+	\item Grow the tree by choosing the node that can be added to the tree by least cost.
+	\item Repeat step 2 until all connected nodes have been visited. If there are still nodes, return to step 1.
+\end{enumerate}
+
+#endif
+\subsection{Applications}
+\begin{itemize}
+	\item Finding a minimum spanning tree
+\end{itemize}
+
+#ifdef hackpackpp
+\subsection{Example Contest Problem: Cow Connection}
+Farmer John has not been very diligent in mowing his farm.
+As a result, the grazing field now resembles a maze with small openings at the intersections.
+
+At night, this maze is unlit, and is very dangerous.
+Farmer John wants to improve safety by installing lights along paths in the maze.
+However, Farmer John does not have much extra money to spend on this effort.
+
+He would like to find a set of paths that connect all of the intersections at minimal cost, but is unsure if he can afford it.
+Please Farmer John, find the cost of lighting the maze and the paths to include.
```

Comment:
This sentence does not very clearly explain what a solution is meant to include. And shouldn't it say something more like "Please help Farmer John to find <something>"?